pkg/systemd/quadlet/quadlet.go

```diff
@@ -1609,6 +1609,9 @@ func handleStorageSource(quadletUnitFile, serviceUnitFile *parser.UnitFile, sour
 	}
 	if source[0] == '/' {
 		// Absolute path
+		if strings.Contains(source, " ") {
+			source = fmt.Sprintf("\"%s\"", source)
+		}
 		serviceUnitFile.Add(UnitGroup, "RequiresMountsFor", source)
```

Comment:
I think that `.Add` should take of quoting the arguments.  I believe there are more such issues which could tackle directly by quoting there.

@ygalblum @alexlarsson WDYT?

Reply:
You could add it to the UnitFile receiver function `Add()`, however, it wouldn't fix the ExecStart weirdness, that'd still have spaces:
```
RequiresMountsFor="/mnt/base part1 part 2" #it's good here

[Service]
#but look at the 'parts' here:
...
ExecStart="/usr/bin/podman run --name=quadlet-demo-mysql --cidfile=%t/%N.cid --replace --rm --cgroups=split --network=systemd-quadlet-demo --sdnotify=conmon -d -v systemd-quadlet-demo-mysql:/var/lib/mysql --mount "type=bind,source=/mnt/base part1 part 2,dst=/mnt" --secret=mysql-root-password-container,type=env,target=MYSQL_ROOT_PASSWORD docker.io/library/mysql:5.6"
```

Reply:
As we can see in the `ExecStart` example, the quotes should not be added automatically

Reply:
Added, and placed unit tests